jerry-debugger/jerry_client.py

```diff
@@ -0,0 +1,352 @@
+        self.debugger.step()
+        self.cont = True
+        self.stop = True
+    do_s = do_step
+
+    def do_backtrace(self, args):
+        """ Get backtrace data from debugger """
+        self.debugger.backtrace(args)
+        self.show = 1
+        self.stop = True
+    do_bt = do_backtrace
+
+    def do_src(self, args):
+        """ Get current source code """
+        if args:
+            line_num = src_check_args(args)
+            if line_num >= 0:
+                print(self.debugger.print_source(line_num, 0))
+            elif line_num == 0:
+                print(self.debugger.print_source(self.debugger.default_viewrange, 0))
+            else:
+                return
+    do_source = do_src
+
+    def do_scroll(self, _):
+        """ Scroll the source up or down """
+        while True:
+            key = sys.stdin.readline()
+            if key == 'w\n':
+                _scroll_direction(self.debugger, "up")
+            elif key == 's\n':
+                _scroll_direction(self.debugger, "down")
+            elif key == 'q\n':
+                break
+            else:
+                print("Invalid key")
+
+    def do_continue(self, _):
+        """ Continue execution """
+        self.debugger.get_continue()
+        self.stop = True
+        self.cont = True
+        if not self.debugger.non_interactive:
+            print("Press enter to stop JavaScript execution.")
+    do_c = do_continue
+
+    def do_finish(self, _):
+        """ Continue running until the current function returns """
+        self.debugger.finish()
+        self.stop = True
+    do_f = do_finish
+
+    def do_dump(self, args):
+        """ Dump all of the debugger data """
+        if args:
+            print("Error: No argument expected")
+        else:
+            pprint(self.debugger.function_list)
+
+    def do_eval(self, args):
+        """ Evaluate JavaScript source code """
+        self.debugger.eval(args)
+        self.stop = True
+        self.cont = True
+    do_e = do_eval
+
+    def do_memstats(self, _):
+        """ Memory statistics """
+        self.debugger.memstats()
+        self.stop = True
+        self.cont = True
+    do_ms = do_memstats
+
+    def do_abort(self, args):
+        """ Throw an exception """
+        self.debugger.abort(args)
+        self.stop = True
+        self.cont = True
+
+    def do_throw(self, args):
+        """ Throw an exception """
+        self.debugger.throw(args)
+        self.stop = True
+        self.cont = True
+
+    def do_exception(self, args):
+        """ Config the exception handler module """
+        print(self.debugger.exception(args))
+
+def _scroll_direction(debugger, direction):
+    """ Helper function for do_scroll """
+    debugger.src_offset_diff = int(max(math.floor(debugger.display / 3), 1))
+    if direction == "up":
+        debugger.src_offset -= debugger.src_offset_diff
+        print(debugger.print_source(debugger.display, debugger.src_offset))
```

Comment:
Put the print functions after the else statement to eliminate the code duplication.